velox/dwio/parquet/thrift/ThriftTransport.h

```diff
@@ -21,12 +21,62 @@
 
 namespace facebook::velox::parquet::thrift {
 
-class ThriftBufferedTransport
-    : public apache::thrift::transport::TVirtualTransport<
-          ThriftBufferedTransport> {
+class ThriftTransport
+    : public apache::thrift::transport::TVirtualTransport<ThriftTransport> {
+ public:
+  virtual uint32_t read(uint8_t* outputBuf, uint32_t len) = 0;
+  virtual ~ThriftTransport() = default;
+};
+
+class ThriftStreamingTransport : public ThriftTransport {
+ public:
+  ThriftStreamingTransport(
+      std::shared_ptr<dwio::common::SeekableInputStream> inputStream,
```

Comment:
As mentioned in https://github.com/facebookincubator/velox/pull/4108#discussion_r1124695215, maybe we should take raw pointer here